accounting/on_chain.go

```diff
@@ -102,48 +183,59 @@ func onChainReportWithPrices(cfg *OnChainConfig, getPrice usdPrice) (Report,
 		isSweep[sweep] = true
 	}
 
-	return onChainReport(
-		filtered, getPrice, openChannels, isSweep, channelOpens,
-		channelCloses,
-	)
+	return onChainReport(info)
 }
 
-// onChainReport produces an on chain transaction report.
-func onChainReport(txns []lndclient.Transaction, priceFunc usdPrice,
-	currentlyOpenChannels map[string]lndclient.ChannelInfo,
-	sweeps map[string]bool, channelOpenTransactions,
-	channelCloseTransactions map[string]lndclient.ClosedChannel) (
-	Report, error) {
+// onChainInformation contains all the information we require to produce an
+// on chain report.
+type onChainInformation struct {
+	txns           []lndclient.Transaction
+	priceFunc      usdPrice
+	sweeps         map[string]bool
+	openedChannels map[string]channelInfo
+	closedChannels map[string]closedChannelInfo
+}
+
+// channelInfo contains information that is common to open and closed channels.
+type channelInfo struct {
+	channelPoint *wire.OutPoint
+	capacity     btcutil.Amount
+	pubKeyBytes  route.Vertex
+	channelID    lnwire.ShortChannelID
+}
+
+// closedChannelInfo contains channel information which has further close info.
+type closedChannelInfo struct {
+	channelInfo
+	closeType      string
+	closeInitiator string
+}
+
+func newChannelInfo(id lnwire.ShortChannelID, chanPoint *wire.OutPoint,
+	pubkey route.Vertex, capacity btcutil.Amount) channelInfo {
 
-	txMap := make(map[string]lndclient.Transaction, len(txns))
-	for _, tx := range txns {
-		txMap[tx.TxHash] = tx
+	return channelInfo{
+		channelID:    id,
+		channelPoint: chanPoint,
+		pubKeyBytes:  pubkey,
+		capacity:     capacity,
 	}
+}
+
+// onChainReport produces an on chain transaction report.
+func onChainReport(info onChainInformation) (
```

Comment:
This could be passed by reference to avoid copying the data. Or the `onChainReport` could even be a a method of `onChainInformation`.